framework/web/Session.php

```diff
@@ -173,10 +173,15 @@ public function close()
     }
 
     /**
-     * Frees all session variables and destroys all data registered to a session.
+     * Frees all session variables and destroys all data registered to a session
+     * even for a closed or a not yet started session.
      */
     public function destroy()
     {
+        // start a previously closed or not yet started session so session_destroy() can handle it
+        if (!$this->getIsActive() && $this->getId()) {
+            @session_start();
+        }
```

Comment:
No it will destroy a current session even it isn't started. To do this `destroy()` will start a session if there is any session identifier.